### ISSUE TITLE
File watcher does not detect a symlinked directory getting re-created

### ISSUE THREAD

**Issue by jfirebaugh** [microsoft/vscode#189427]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.80.2
- OS Version: macOS 13.4.1

Steps to Reproduce:

1. Run the following:
   ```
   mkdir -p scratch/src
   mkdir scratch2
   dst=$(cd scratch2 && pwd)
   ln -s $dst scratch/src/symlink
   echo "import { Foo } from './symlink/index'" >scratch/src/test.ts
   echo "export const Foo = 1;" >scratch2/index.ts
   code --disable-extensions scratch
   ```
   Open `test.ts`. It should have no errors.
3.  Run `rm -rf scratch2`. After vscode detects the change, it should display a red squiggle under `./symlink/index` with the error "Cannot find module ...".
4. Run `mkdir scratch2 && echo "export const Foo = 1;" >scratch2/index.ts`

The expected behavior at this point is that vscode recognizes that the target directory of the symlink once again exists, and removes the red squiggle.

The actual behavior is that the squiggle does not go away. You must close and reopen the project to get vscode to recognize that the code is valid.

**Comment by jfirebaugh** [comment]
This applies for for symlinked files as well: you can replace step 2 with `rm scratch2/index.ts`, and observe the same actual behavior after recreating that file in step 3.

After step 3, command clicking on the text with the squiggle opens a tab that displays the following:

![Screenshot 2023-08-01 at 1 08 01 PM](https://github.com/microsoft/vscode/assets/98601/f236e92f-c3f0-4229-b723-cbe4568778e5)

However, you can open `src/symlink/index.ts` successfully from the Explorer.

**Comment by bpasero** [comment]
Our file watcher does not support changes to targets that are in the workspace behind symbolic links. In other words, the file watcher will not automatically follow symlinks for performance reasons.

We have limited support to add additional folders to be watched via the `files.watcherInclude` setting, but in this scenario when I test, I am not seeing file events for when the folder is being recreated.

However here I believe TypeScript is doing something on their own, because they do detect the delete but not the recreation, so maybe this is something TypeScript deals specifically.

**Comment by jfirebaugh** [comment]
Yes, I concur that it's a TypeScript issue -- I was able to reproduce the same issue in another editor (IntelliJ IDEA). Do you have the ability to transfer the issue to the microsoft/typescript repository?
